data_model/src/asset.rs

```diff
@@ -416,8 +416,24 @@ pub struct NewAssetDefinition {
     metadata: Metadata,
 }
 
+impl Builder for NewAssetDefinition {
+    type Target = AssetDefinition;
+
+    #[cfg(feature = "mutable_api")]
```

Comment:
I think you should place it above the impl block

Reply:
My intention to capture relation between builder and target type. 
To be able to express smt like this:
```rust
impl Identifiable for NewRole {
    type Id =  <<Self as Builder>::Target as Identifiable>::Id;
}
```
If we make `Builder` trait available only under "mutable_api" feature we lose this ability.
I can do it both ways, what do you think?

Reply:
```rs
impl Identifiable for NewRole { 
    type Id = <<Self as Builder>::Target as Identifiable>::Id; 
}
```
is essentially the same thing as
```rs
impl Identifiable for NewRole { 
    type Id = <Role as Identifiable>::Id; 
}
```
after the compiler fills in the types

Reply:
Ok, i will make `Builder` only available under "mutable_api", thank you for the suggestion.